lib/resourcebuilder/podspec.go

```diff
@@ -22,19 +19,23 @@ func updatePodSpecWithProxy(podSpec *corev1.PodSpec, containerNames []string, ht
 			}
 			found = true
 
-			podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: httpProxy})
-			podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "HTTPS_PROXY", Value: httpsProxy})
-			podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "NO_PROXY", Value: noProxy})
+			if hasProxy {
+				podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: httpProxy})
+				podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "HTTPS_PROXY", Value: httpsProxy})
+				podSpec.Containers[i].Env = append(podSpec.Containers[i].Env, corev1.EnvVar{Name: "NO_PROXY", Value: noProxy})
+			}
 		}
 		for i := range podSpec.InitContainers {
 			if podSpec.InitContainers[i].Name != containerName {
 				continue
 			}
 			found = true
 
-			podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: httpProxy})
-			podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "HTTPS_PROXY", Value: httpsProxy})
-			podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "NO_PROXY", Value: noProxy})
+			if hasProxy {
+				podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "HTTP_PROXY", Value: httpProxy})
+				podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "HTTPS_PROXY", Value: httpsProxy})
+				podSpec.InitContainers[i].Env = append(podSpec.InitContainers[i].Env, corev1.EnvVar{Name: "NO_PROXY", Value: noProxy})
+			}
 		}
 
 		if !found {
```

Comment:
Trying to test with Cluster Bot `launch 4.16,openshift/cluster-version-operator#1014,openshift/console-operator#832 gcp` ([logs][1]), the cluster installed fine, and I see no sign of the expected line in CVO logs:

```console
$ oc -n openshift-cluster-version logs -l k8s-app=cluster-version-operator --tail -1 | grep -c 'requested injection for non-existent container'
0
```

Not yet clear to me what's going on.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-gcp-modern/1742631262188736512

Reply:
@wking I tested it with a new PR which has busted manifest, and launch command failed, my launch command is:
`launch 4.16,openshift/cluster-version-operator#1014,openshift/console-operator#874 gcp`

full log can be found in https://prow.ci.openshift.org/view/gs/test-platform-results/logs/release-openshift-origin-installer-launch-gcp-modern/1763113136023408640